drivers/bus/isapnp/isapnpres.h

```diff
@@ -59,23 +56,35 @@ typedef struct _ISAPNP_COMPATIBLE_ID_ENTRY
     LIST_ENTRY IdLink;
 } ISAPNP_COMPATIBLE_ID_ENTRY, *PISAPNP_COMPATIBLE_ID_ENTRY;
 
-typedef struct _ISAPNP_ALTERNATIVES
+typedef enum
+{
+    dfNotStarted,
+    dfStarted,
+    dfDone
+} ISAPNP_DEPENDENT_FUNCTION_STATE;
+
+typedef struct _ISAPNP_RESOURCE
 {
-    ISAPNP_IO_DESCRIPTION Io[ISAPNP_MAX_ALTERNATIVES];
-    ISAPNP_IRQ_DESCRIPTION Irq[ISAPNP_MAX_ALTERNATIVES];
-    ISAPNP_DMA_DESCRIPTION Dma[ISAPNP_MAX_ALTERNATIVES];
-    ISAPNP_MEMRANGE_DESCRIPTION MemRange[ISAPNP_MAX_ALTERNATIVES];
-    ISAPNP_MEMRANGE32_DESCRIPTION MemRange32[ISAPNP_MAX_ALTERNATIVES];
-    UCHAR Priority[ISAPNP_MAX_ALTERNATIVES];
-    UCHAR IoIndex;
-    UCHAR IrqIndex;
-    UCHAR DmaIndex;
-    UCHAR MemRangeIndex;
-    UCHAR MemRange32Index;
-
-    _Field_range_(0, ISAPNP_MAX_ALTERNATIVES)
-    UCHAR Count;
-} ISAPNP_ALTERNATIVES, *PISAPNP_ALTERNATIVES;
+    UCHAR Type;
+#define ISAPNP_RESOURCE_TYPE_END               0
```

Comment:
I wonder whether you could turn these into an enum `ISAPNP_RESOURCE_TYPE` :thinking:

Reply:
I want to force it to be 1-byte in size

Reply:
Unfortunately this adds 6 extra bytes into the structure (18 -> 24).